src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -593,7 +593,25 @@ public synchronized void applyDML(Sequenced<DMLEvent> sequencedEvent) throws Exc
       .setTableName(normalizedTableName)
       .build();
     long sequenceNumber = sequencedEvent.getSequenceNumber();
-    gcsWriter.write(new Sequenced<>(normalizedDMLEvent, sequenceNumber));
+    try {
+      Failsafe.with(new RetryPolicy<>()
+                      .withMaxAttempts(25)
+                      .withMaxDuration(Duration.of(2, ChronoUnit.MINUTES))
+                      .withBackoff(1, 30, ChronoUnit.SECONDS)
+                      .withJitter(0.1))
+              .run(() -> gcsWriter.write(new Sequenced<>(normalizedDMLEvent, sequenceNumber)));
+    } catch (TimeoutExceededException e) {
+      // if the retry timeout was reached, throw a DeltaFailureException to fail the pipeline immediately
+      DeltaFailureException exc = new DeltaFailureException
+              (String.format("Unable to write event %s to bytes.", event), e);
+      flushException = exc;
+      throw exc;
```

Comment:
Let skip catching TimeoutExceededException, as we don't want to fail the pipeline permanently but the let worker retry

Reply:
Have updated the code.Both timeout and interrupted exception handling isn't required.